types/react-i18next/src/I18n.d.ts

```diff
@@ -0,0 +1,15 @@
+import * as React from "react";
+import { i18n } from "i18next";
+
+export interface i18nProps {
+    wait?: boolean;
+    ns: string | string[];
+    nsMode?: string;
+    bindI18n?: string;
+    bindStore?: string;
+    i18n?: i18n;
+    initialI18nStore?: any;
+    initialLanguage?: string;
```

Comment:
you're missing here very important part for type checking -> annotation for type safe children.

Please add 

```ts
interface Options {
  i18n: i18n
  t: TranslationFunction
}

interface i18nProps {
  // ...
  children: (t: TranslationFunction, options: Options) => JSX.Element
}
```

this will make TS infer children function, so consumer will get all intellisense and type checking without the need to explicitly annotate child function.

then simplify test component as well:

```ts
<I18n>
 ( t, {i18n} ) => (
             <div>
                 <h1>{t('keyFromDefault')}</h1>
                 <p>{t('anotherNamespace:key.from.another.namespace', {/* options t options */})}</p>
             </div>
         )
</I18n>
```

thx, would love to get this PR in so don't have to override types in every project :)

Reply:
Thanks for your additions, have added/adapted the type definitions accordingly